### ISSUE TITLE
Rule: Prefer advanced path resolution in TypeScript 2.0

### ISSUE THREAD

**Issue by michaeljota** [palantir/tslint#1809]
### Request Rule Report

- __TSLint version__: `4.0.1`
- __TypeScript version__: `2.0.3`
- __Running TSLint via__: CLI / VS Code

#### Rule description

This rule should check if there is a `baseUrl`, with or without any `paths`, set in compiler options in the tsconfig file. If they are, this should suggest the imports of files that are not children of the current folder, directly from the `baseUrl` setting.

#### Actual behavior

There is not rule implemented anywhere, as I know. 

#### Reasoning:

The advance path resolution simplifies the import of other files, regardless the distance of the two modules in the tree folder. Also, it made the code simpler and more readable, cause the lack of `'./../../../../'` in the imports. 

Instead of 
```ts
import { SomeAwesomeClass } from './../../../../some-folder-in-base-url/some-other-folder/class-file'
```
We could write as
```ts
import { SomeAwesomeClass } from 'some-other-folder/class-file'
```

Cause the advance path resolution, will resolve `some-other-folder` from `some-folder-in-base-url`.

**Comment by marcelh-gh** [comment]
hi,

is there any progress about this?
The absence of a solution leads to lots of false errors when baseURL and paths options are used.

Marcel

**Comment by sergebat** [comment]
I'd definitely like to see such rule. Let me add that there's somewhat similar rule in https://github.com/Microsoft/tslint-microsoft-contrib (no-relative-imports), but it completely forbids local import, which might be too strong limit.

**Comment by JoshuaKGoldberg** [comment]
Indeed, `tslint-microsoft-contrib`'s `no-relative-import` is the way to go there. Closing this as a subset of #1142.
